Add test plan for Quarkus-728 (#28)
quarkus-qe/quarkus-test-plans: QUARKUS-728.md

# Quarkus-728 - Add support for OpenJDK 17

JIRA link: https://issues.redhat.com/browse/QUARKUS-728

The goal is to verify OpenJDK 17 integration

## Scope of the testing
Test development will focus on  

>  - New specifict scenarios will be created to cover the integration of new objects as “sealed” or “records” classes with several extensions as Hibernate or RestEasy.

Reviewer: RESTEasy

Reviewer: @pjgg not updated, still see `RestEasy` instead of `RESTEasy`

>  - Current escenarios must work with Java 17  

Reviewer: ```suggestion
 - Current scenarios must work with Java 17  
```

> 

Reviewer: - S2I needs to work with Java 17 project

### Impact on testsuites and testing automation:

>  - Ensure this coverage works both in JVM and NATIVE mode 

Reviewer: NATIVE supports just Java 11, adjust this line and make a note about NATIVE

 - Ensure this coverage works in Openshift (s2i)
 - New test suite will be created in order to develop specifict Java 17 scenarios
 - New test suite will use Gradle as a build tool

>     

Reviewer: No information about impact on 
 - jobs definition
 - test execution time
 - lab capacity
 - test cycle length

Will all the Java 11 jobs have Java 17 variants?

Where will be the Java installation taken from, is RHEL going to support it via RPMs? How will be the tooling for VMs preparation changed to cover new JDK on Jenkins slaves, Jenkins master tools config will need some small Java 17 related adjustments too.

Author: @mjurc could you help me to clarify which Jobs should support Java17?

Reviewer: Please check Testing strategy document for RHBQ 2.x, it speaks about strategy for platforms, the idea is to have primary one and subset of coverage for other platforms.

The same applies to Java 11 and Java 17. Java 11 should be the primary version, subset of coverage for Java 17 and over time (assume 1-2 years) we switch Java 17 as the primary platform and Java 11 as the subsequent one.

For Java 17 we need to execute the jobs with the new TS for sure + Java 17 specific TS + Quickstarts (with redefined source / target to 17 via command line). Maybe also StartStopTS should be executed with Java 17.

Author: I have re-write all of this section following your comments.

# Quarkus-728 - Add support for OpenJDK 17

JIRA link: https://issues.redhat.com/browse/QUARKUS-728

The goal is to verify OpenJDK 17 integration

## Scope of the testing
Test development will focus on  
 - New specifict scenarios will be created to cover the integration of new objects as “sealed” or “records” classes with several extensions as Hibernate or RESTEasy.
 - Current scenarios must work with Java 17

>  - S2i (source to image) must works with Java 17  

Reviewer: S2i => S2I

### Impact on testsuites and testing automation:

>  - Ensure this coverage works in JVM mode

Reviewer: Please add subsection or new bullet point for enhancing GitHub Actions to cover both Java 11 and 17 in our "quarkus-qe" testsuites

 - NATIVE mode is not required in Java 17 

>  - Ensure this coverage works in Openshift (s2i)

Reviewer: S2I

 - New test suite will be created in order to develop specifict Java 17 scenarios
 - New test suite will use Gradle as a build tool
 - Following RHBQ 2.x testing strategy Java 11 will be our primary platform that will support all the defined platforms, but Java 17 must also support a subsequent of these platforms:
   - RHEL 8 x86_64 with OpenJDK 17
   - Windows Server 2019 x86_64 with OpenJDK 17
   - OpenShift 3.11 with UBI8 OpenJDK 17
   - OpenShift 4.latest with UBI8 OpenJDK 17
   - OpenShift 4.LTS with UBI8 OpenJDK 17
   - OpenShift OSD with UBI8 OpenJDK 17
   - RHEL 9 x86_64 with OpenJDK 17

>    - RHEL 9 x86_64 with Mandrel 21

Reviewer: drop this line

# Quarkus-728 - Add support for OpenJDK 17

JIRA link: https://issues.redhat.com/browse/QUARKUS-728

The goal is to verify OpenJDK 17 integration

## Scope of the testing
Test development will focus on  
 - New specifict scenarios will be created to cover the integration of new objects as “sealed” or “records” classes with several extensions as Hibernate or RestEasy.
 - Current scenarios must work with Java 17
 - S2i (source to image) must works with Java 17  

### Impact on testsuites and testing automation:
 - Ensure this coverage works in JVM mode
 - NATIVE mode is not required in Java 17 
 - Ensure this coverage works in Openshift (s2i)
 - New test suite will be created in order to develop specifict Java 17 scenarios
 - New test suite will use Gradle as a build tool

>  - The following Jenkins jobs must be duplicated to support Java 17:

Reviewer: ```suggestion
 - The following Jenkins jobs must be duplicated or converted to matrix jobs to support Java 17:
```

>    - RHBQ

Reviewer: move the ones with `?` into separate section and mention that we will need to decide in the wider QE team discussion if these jobs need Java 17 equivalent or not

Author: I have re-written all of this section, but I add a note, where I explain that the final Jenkins job list must be discussed by the QE team.

     - rhbq-1.11-rhel8-jdk11-startstop-ts-code-quarkus
     - rhbq-1.11-rhel8-jdk11-startstop-ts-generator
     - rhbq-1.11-rhel8-jdk11-startstop-ts-special-chars
     - rhbq-1.11-rhel8-jdk11-workshop-ts ?
     - rhbq-1.11-win19-jdk11-beefy-ts-jvm ?
     - rhbq-1.11-win19-jdk11-openshift-ts-jvm-ocp-4-stable-deployment-strategy ?
     - rhbq-1.11-win19-jdk11-quickstarts-ts-jvm-quarkus-bom ?
     - rhbq-1.11-win19-jdk11-quickstarts-ts-jvm-quarkus-universe-bom ?
     - rhbq-1.11-win19-jdk11-startstop-ts-code-quarkus ?
     - rhbq-1.11-win19-jdk11-startstop-ts-special-chars ?
   - Upstream
     - quarkus-main-rhel8-jdk11-database-ts-jvm ?
     - quarkus-main-rhel8-jdk11-extensions-combinations-ts
     - quarkus-main-rhel8-jdk11-openshift-new-ts-jvm ?
     - quarkus-main-rhel8-jdk11-openshift-new-ts-jvm-githubci ?
     - quarkus-main-rhel8-jdk11-openshift-tf-jvm-githubci ?
     - quarkus-main-rhel8-jdk11-openshift-ts-jvm
     - quarkus-main-rhel8-jdk11-openshift-ts-jvm-githubci
     - quarkus-main-rhel8-jdk11-quickstarts-ts-jvm
     - quarkus-main-rhel8-jdk17-beefy-ts-jvm
     - quarkus-main-rhel8-jdk17-quickstarts-ts-jvm
     - quarkus-startstop-perflab ?

>  - Lab capacity should be increased in a 50% in terms of CPU and RAM memory

Reviewer: ```suggestion
 - Lab capacity should be ideally increased by 50% in terms of CPU and RAM memory, if that doesn't happen the execution time of testing will be prolonged, almost doubled. 
```

>  - New Jenkins slaves that will run Java17 pipelines will have OpenJDK 17 installed preferably an RHEL OpenJDK distribution but if doesn't exist because is an early stage version then we will use an SDKman distribution

Reviewer: Java17 => Java 17
